File: app/src/main/java/ru/sicampus/bootcamp2025/ui/map/MapFragment.java
package ru.sicampus.bootcamp2025.ui.map;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.google.android.gms.maps.SupportMapFragment;
import com.google.android.gms.maps.model.LatLng;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import ru.sicampus.bootcamp2025.R;
import ru.sicampus.bootcamp2025.databinding.FragmentMapBinding;
import ru.sicampus.bootcamp2025.databinding.FragmentProfileBinding;
import ru.sicampus.bootcamp2025.domain.entites.FullUserEntity;
import ru.sicampus.bootcamp2025.domain.entites.FullVolunteerCenterEntity;
import ru.sicampus.bootcamp2025.domain.entites.ItemVolunteerCenterEntity;
import ru.sicampus.bootcamp2025.ui.profile.MapViewModel;
import ru.sicampus.bootcamp2025.ui.utils.Utils;

public class MapFragment extends Fragment {

    private static final String KEY_ID = "id";

    private FragmentMapBinding binding;
    private MapViewModel viewModel;
    private static List<ItemVolunteerCenterEntity> volunteerCenters = new ArrayList<>();
    //private static List<Place> places = new ArrayList<>();
    public MapFragment() {
        super(R.layout.fragment_map);
    }

//    public getViewFr() {
//        return
//    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding = FragmentMapBinding.bind(view);
        viewModel = new ViewModelProvider(this).get(MapViewModel.class);
        viewModel.stateLiveData.observe(getViewLifecycleOwner(), state -> {
            if (state.isLoading() == false) {
                if (state.getItems() != null) {
                    volunteerCenters = state.getItems();
                    /*places = new ArrayList<>();
                    for (ItemVolunteerCenterEntity vc : volunteerCenters) {
                        Place place = new Place(vc.getId(), new LatLng(Double.parseDouble(vc.getLatitude()), Double.parseDouble(vc.getLongitude())));
                        places.add(place);
                    }*/

                    //int actionId = getNavigationMenuItemId();

                    //view.getRootView().findViewById(R.id.nav_view).setVisibility(View.VISIBLE);

//                    BottomNavigationView navigationView = view.getRootView().findViewById(R.id.nav_view); // selectBottomNavigationBarItem(actionId);
//                    navigationView.setSelectedItemId(R.id.list);

                    SupportMapFragment supportMapFragment = (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.map_view); // getParentFragmentManager().findFragmentById(R.id.map_view);

                    supportMapFragment.getMapAsync(new MapService(this.getContext(), view));
                } else {
                    Toast.makeText(this.getContext(), "Error", Toast.LENGTH_SHORT).show();
                }
            }
        });

        //Integer id = getArguments() != null ? getArguments().getInt(KEY_ID) : null;
        //if (id == null) throw new IllegalStateException("ID is null");

        viewModel.load();

    }
    public static List<ItemVolunteerCenterEntity> getVolunteerCenters(){
        return volunteerCenters;
    }
    public static ItemVolunteerCenterEntity getVolunteerCentersName(String name){
        for(ItemVolunteerCenterEntity p: volunteerCenters){
            if (Objects.equals(name, p.getId().toString())){
                return p;
            }
        }
        return null;
    }
    @Override
    public void onDestroyView() {
        binding = null;
        super.onDestroyView();
    }

    public static Bundle getBundle(@NonNull String id) {
        Bundle bundle = new Bundle();
        bundle.putString(KEY_ID, id);
        return bundle;
    }
}

File: app/src/main/java/ru/sicampus/bootcamp2025/ui/map/MapFragment.kt
package ru.sicampus.bootcamp2025.ui.map

import android.graphics.drawable.Drawable
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.viewModelScope
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.GlideException
import com.bumptech.glide.request.RequestListener
import com.google.android.gms.maps.CameraUpdateFactory
import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.OnMapReadyCallback
import com.google.android.gms.maps.SupportMapFragment
import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.Marker
import com.google.android.gms.maps.model.MarkerOptions
import com.google.android.material.bottomsheet.BottomSheetDialog
import kotlinx.coroutines.launch
import ru.sicampus.bootcamp2025.R
import ru.sicampus.bootcamp2025.domain.map.DepartmentEntity
import ru.sicampus.bootcamp2025.domain.map.PlaceEntity


class MapFragment() : Fragment(R.layout.fragment_map),
    OnMapReadyCallback,
    GoogleMap.OnMapClickListener,
    GoogleMap.OnMapLongClickListener,
    GoogleMap.OnMarkerClickListener {

    private lateinit var googleMap: GoogleMap
    private val viewModel: MapViewModel by viewModels() { MapViewModel.Factory}

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val mapFragment = childFragmentManager.findFragmentById(R.id.google_map) as? SupportMapFragment
        if (mapFragment == null) {
            Log.e("MapFragment", "MapFragment is null!")
        } else {
            mapFragment.getMapAsync(this)
        }

        viewModel.placesLiveData.observe(viewLifecycleOwner) { departments ->

            if (::googleMap.isInitialized) {
                for (department in departments) {
                    googleMap.addMarker(MarkerOptions().position(department.place.latLng).title(department.name))
                }
            }
        }
        viewModel.selectedDepartment.observe(viewLifecycleOwner) { department ->
            if (department != null) {
                showPlaceDetails(department)
            } else {
                Toast.makeText(requireContext(), "Place not found", Toast.LENGTH_SHORT).show()
            }
        }
    }

    override fun onMapReady(map: GoogleMap) {
        googleMap = map
        googleMap.setOnMapClickListener(this)
        googleMap.setOnMapLongClickListener(this)
        googleMap.setOnMarkerClickListener(this)

        viewModel.getPlaces()

    }

    override fun onMapClick(latLng: LatLng) {
        Toast.makeText(requireContext(), "Coords: ${latLng.latitude} ${latLng.longitude}", Toast.LENGTH_SHORT).show()
    }

    override fun onMapLongClick(latLng: LatLng) {
        Toast.makeText(requireContext(), "LONG Coords: ${latLng.latitude} ${latLng.longitude}", Toast.LENGTH_SHORT).show()
    }

    override fun onMarkerClick(marker: Marker): Boolean {
        viewModel.getDepartmentByName(marker.title)
        return false
    }

    private fun showPlaceDetails(department: DepartmentEntity) {
        val place = department.place
        val dialog = BottomSheetDialog(requireContext())
        dialog.setContentView(R.layout.bottom_sheet_dialog)
        dialog.window?.apply {
            setBackgroundDrawableResource(android.R.color.transparent)
        }
        dialog.show()
        Log.d("MapFragment", "${place.pathToImage}")
        dialog.findViewById<ImageView>(R.id.image)?.let {
            Glide.with(this)
                .load(place.pathToImage) // TODO()
                .placeholder(R.drawable.ic_photo)
                .error(R.drawable.ic_back)
                .into(it)
        }
//        dialog.findViewById<ImageView>(R.id.image)?.let {
//            imageService.setImage(it, place.pathToImage)
//        }
        dialog.findViewById<TextView>(R.id.name)?.text = place.name
        dialog.findViewById<TextView>(R.id.address)?.text = place.address
        dialog.findViewById<TextView>(R.id.description)?.text = place.information
        dialog.findViewById<TextView>(R.id.attach)?.setOnClickListener {
            viewModel.changeDepartmentAttach(department.name)
        }
    }

}
